### PR TITLE
fix: Adapted transform_tracking_logs command for enhanced CLI integration

### PR DESCRIPTION
**Pull request by nandodev-net**
### **Adaptation of _transform_tracking_logs_ Command**

**Description:**

This pull request introduces a series of click options to the transform_tracking_logs command, enhancing its versatility and user-friendliness.

**Objective:**

The primary goal of these additions is to provide a clearer and more structured way for users to interact with the transform_tracking_logs command. By allowing explicit configuration via command-line options, we make the functionality more transparent and easier to customize.

Changes Made:

- Refactored to utilize kwargs, simplifying the command signature.
- Enhanced readability by breaking long input into many @click.option.

**How to Test:**

The command can be tested by running something like this:

```
`tutor k8s do transform-tracking-logs \
    --source_provider LOCAL \
    --source_config '{"key": "/openedx/data/", "container": "logs", "prefix": "tracking.log"}' \
    --destination_provider LRS \
    --transformer_type xapi \
    --batch_size 5000 \
    --sleep_between_batches_secs 5.0 \
    --dry_run`
```

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @nandodev-net!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: tutoraspects/commands_v0.py [hunk 1]
```diff
@@ -110,18 +110,83 @@ def dump_courses_to_clickhouse(context, options) -> None:
...
+        f"--source_provider {kwargs['source_provider']} "
+        f"--source_config {kwargs['source_config']} "
+        f"--destination_provider {kwargs['destination_provider']} "
+        f"--transformer_type {kwargs['transformer_type']}"
+    )    
+    
+    if 'destination_config' in kwargs and kwargs['destination_config']:
+        options += f" --destination_config {kwargs['destination_config']}"
+        
+    options += f" --batch_size {kwargs['batch_size']} --sleep_between_batches_secs {kwargs['sleep_between_batches_secs']}"
```
Comment (Ian2012) [inline tutoraspects/commands_v0.py:183]: I would prefer to validate every input using something like:

```python
for arg in kwargs:
    if kwargs[arg]:
        options.append(f"--{arg} {kwargs[arg]}")
```

This way, only defined values are passed for the downstream command without any intervenient from the wrapper

File: tutoraspects/commands_v0.py [hunk 2]
```diff
@@ -110,18 +110,83 @@ def dump_courses_to_clickhouse(context, options) -> None:
...
+
+    if 'dry_run' in kwargs and kwargs['dry_run']:
+        options += " --dry_run"
+
+    command = f"./manage.py lms transform_tracking_logs {options}"
```
Comment (Ian2012) [inline tutoraspects/commands_v0.py:188]: I would keep this

**Review by Ian2012 — CHANGES_REQUESTED** [review]

File: tutoraspects/commands_v0.py [hunk 3]
```diff
@@ -110,18 +110,78 @@ def dump_courses_to_clickhouse(context, options) -> None:
...
-    command = f"""
-    ./manage.py lms transform_tracking_logs {options}
-    """
+    options = []
+    for arg, value in kwargs.items():
+        if value:
+            options.append(f"--{arg} {value}")
+
+    if 'dry_run' in kwargs and kwargs['dry_run']:
+        options.append("--dry_run")
```
Comment (Ian2012) [inline tutoraspects/commands_v0.py:179]:
```suggestion
        if not value:
            continue
        if arg == 'dry_run':
            options.append(f"--{arg}")
        elif arg == 'source_config' or arg == 'destination_config':
            options.append(f"--{arg} '{value}'")
        else:
            options.append(f"--{arg} {value}")
```

**Review by Ian2012 — CHANGES_REQUESTED** [review]
Just some patches for some inputs, thanks for your contribution.

Once this is addressed, this will be approved

**Comment by Ian2012** [comment]
Fixes: https://github.com/openedx/tutor-contrib-aspects/issues/208

**Review by Ian2012 — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@nandodev-net 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/tutor-contrib-aspects&entry.867055334=https://github.com/openedx/tutor-contrib-aspects/pull/291&entry.1484655318=https://github.com/nandodev-net&entry.752974735=2023-08-08+22:07&entry.1917517419=2023-08-14+21:50&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**